sql/statement_execution.go

```diff
@@ -0,0 +1,117 @@
+package sql
```

Comment:
Can you use the GoSDK for this? See https://github.com/databricks/databricks-sdk-go/blob/main/service/sql/api.go which already defines StatementExecutionAPI and WarehousesAPI.

Reply:
Way better, thanks for the recommendation! I've made the suggested changes.

Reply:
Since ExactlyOneOf has this behavior:
>  It will return a validation error if none are specified as well.

And the current behavior of this resource is that `cluster_id` is _optional_ (if not specified, a default cluster will be created named "terraform-sql-table") I believe the most correct constraint is your first suggestion:
```go
s["cluster_id"].ConflictsWith = []string{"warehouse_id"}
s["warehouse_id"].ConflictsWith = []string{"cluster_id"}
```